File: UICollectionViewLayout/uicollectionview-gridlayout-master/Examples/Sticky Headers/Sticky Headers/HeaderView.m
//
//  HeaderView.m
//  Sticky Headers
//
//  Created by Tim Moose on 7/10/13.
//  Copyright (c) 2013 tmoose@vast.com. All rights reserved.
//

#import "HeaderView.h"

@implementation HeaderView

- (id)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        
        _titleLabel = [[UILabel alloc] initWithFrame:self.bounds];
        _titleLabel.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
        _titleLabel.backgroundColor = [UIColor clearColor];
        [self addSubview:_titleLabel];
    }
    
    return self;
}

@end

File: Examples/UICollectionViewDemo/UICollectionViewDemo/CalendarView/HeaderView.swift
//
//  HeaderView.swift
//  UICollectionViewDemo
//
//  Created by 伯驹 黄 on 2017/6/1.
//  Copyright © 2017年 伯驹 黄. All rights reserved.
//

import UIKit

class HeaderView: UICollectionReusableView {
    let titleLabel = UILabel()
    
    let line = CALayer()
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        titleLabel.frame = CGRect(x: 10, y: 10, width: frame.width - 20, height: 21)

        addSubview(titleLabel)

        line.backgroundColor = UIColor(red: 127 / 255, green: 127 / 255, blue: 127 / 255, alpha: 1).cgColor
        layer.addSublayer(line)
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        line.frame = CGRect(x: 0, y: frame.height - 0.5, width: frame.width, height: 0.5)
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}
